src/server/game/Spells/Spell.cpp

```diff
@@ -2328,7 +2328,26 @@ void Spell::TargetInfo::DoTargetSpellHit(Spell* spell, uint8 effIndex)
         return;                                             // No missinfo in that case
 
     if (_spellHitTarget)
+    {
```

Comment:
why did you change the order here?

Reply:
Because function  

void Spell::TargetInfo::DoDamageAndTriggers(Spell* spell)

rreturn in LINE when spell casted bu GO

// Get original caster (if exist) and calculate damage/healing from him data
     // Skip if m_originalCaster not available
     Unit* caster = spell->m_originalCaster ? spell->m_originalCaster : spell->m_caster->ToUnit();
     if (!caster)
         return;

So AI functions for gobj casts not called

Reply:
No, this is a bad change, `Spell::TargetInfo::DoTargetSpellHit` is called for each effect on target but you only want to call AI hooks once